### DISCUSSION TITLE
Not getting cluster names in the dataframe after pathway expression

### DISCUSSION THREAD

**Discussion by jjyotikataria** [reactome/ReactomeGSA#30] [Q&A] [answered]
I ran the following commands after taking a clustered Seurat object.

gsva_result <- analyse_sc_clusters(pbmc, verbose = TRUE)

pathway_expression <- pathways(gsva_result)

colnames(pathway_expression) <- gsub("\\.Seurat", "", colnames(pathway_expression))

In the example, the heatmap shows Cluster1 and Cluster 2 etc, in my plot it shows X0 and X1 for the two clusters. Could you please help me with it? @nturaga @hawrobin @teslajoy @jweiser @SolomonShorser-OICR

**Comment by jgriss** [comment, marked as answer]
Hi @jjyotikataria,

Thanks for your interest in ReactomeGSA!

The resulting names depend on your cluster names or rather the primary identifier used. For Seurat, this is done using the `Idents` function.

In the example, the dataset contained "Cluster 1" ... as the main `Ident`.

Once you adapt yours to match your cluster assignments, it should work.

Kind regards,
Johannes

Btw @nturaga this is the best place to get support for ReactomeGSA
